state/protocol/badger/validity.go

```diff
@@ -377,11 +377,27 @@ func SanityCheckConsensusNodeRootSnapshotValidity(snapshot protocol.Snapshot) er
 		return fmt.Errorf("could not query spork root block height: %w", err)
 	}
 
-	cfg := DefaultConfig()
-	if head.Height+sporkRootBlockHeight < cfg.transactionExpiry {
-		return fmt.Errorf("invalid root snapshot length, expecting at least %d, got %d",
-			cfg.transactionExpiry, head.Height-sporkRootBlockHeight)
+	sealingSegment, err := snapshot.SealingSegment()
+	if err != nil {
+		return fmt.Errorf("could not query sealing segment: %w", err)
 	}
 
+	sealingSegmentLength := uint64(len(sealingSegment.AllBlocks()))
+	cfg := DefaultConfig()
```

Comment:
By using the default config rather than a config instance actually used by the protocol state instance, we're already assuming that the `transactionExpiry` value is constant. I think this would be clearer if we just use the constant value directly: `flow.DefaultTransactionExpiry`.

(The same argument holds for the `Config` type as well. As-is, it is not safe to configure any value for `config.transactionExpiry` besides the global constant, so it would be clearer to use the constant directly.)

Reply:
Yeah, agree, I have removed config at all since we use the default one all the time. Will bring it back when it's needed.